CMakeLists.txt

```diff
@@ -1,13 +1,36 @@
-# This file is used for testing only
+project(editorconfig-core-qt)
+cmake_minimum_required(VERSION 3.20)
 
-# To perform the test, run `cmake .` at the root of the project tree followed
-# by ctest .
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+set(CMAKE_AUTOUIC ON)
 
-cmake_minimum_required(VERSION 2.6)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-# Do not check any compiler
-project(editorconfig-core-qt NONE)
+find_package(Qt5 COMPONENTS Core Gui Network REQUIRED)
 
 enable_testing()
-set(EDITORCONFIG_CMD ${CMAKE_BINARY_DIR}/editorconfig-app.exe)
+set(EDITORCONFIG_CMD ${CMAKE_BINARY_DIR}/src/editorconfig-app/editorconfig-app)
 add_subdirectory(tests)
+
+if (BUILD_STATIC)
+    set(CMAKE_STATIC_LIBRARY_SUFFIX "_static${CMAKE_STATIC_LIBRARY_SUFFIX}")
+    add_library(editorconfig-core-qt STATIC
+        EditorConfig.cpp
+        EditorConfig.h
+    )
+else()
+    add_library(editorconfig-core-qt SHARED
+        EditorConfig.cpp
+        EditorConfig.h
+    )
+endif()
+
+target_link_libraries(editorconfig-core-qt PRIVATE Qt5::Core )
```

Comment:
Qt6 support missing.

Reply:
We can always add it in later or add some if/else check in cmake to see which library to link. Not a big deal right now.

Reply:
OK, another PR will come eventually :)